taichi/program/program.h

```diff
@@ -121,22 +115,9 @@ class TI_DLL_EXPORT Program {
 
   ~Program();
 
-  CompileConfig &this_thread_config() {
-    // std::unordered_map is not thread safe even if we do the rehash in
-    // advance, so we need to add a lock to protect it.
-    std::shared_lock<std::shared_mutex> read_lock(config_map_mut);
-    auto thread_id = std::this_thread::get_id();
-    if (!configs.count(thread_id)) {
-      read_lock.unlock();
-      std::unique_lock<std::shared_mutex> write_lock(config_map_mut);
-      configs[thread_id] = configs[main_thread_id_];
-      return configs[thread_id];
-    }
-    return configs[thread_id];
-  }
-
-  const CompileConfig &config() {
-    return configs[main_thread_id_];
+  // TODO(PGZXB): Return `const CompileConfig &` (blocked by some tests)
+  CompileConfig &compile_config() {
```

Comment:
nit: is it possible that we have another method that returns a const reference, and make sure only the problematic tests use this non-const version? Thanks!

Reply:
Maybe we can provide `Program::set_compile_config`(only for tests)?
p.s. And I will refactor those tests to remove the temporary `Program::set_compile_config`.

Reply:
sg!

Reply:
I found that the dependencies on the non-const version in tests can be removed directly, so I killed them in this PR and fixed the ` CompileConfig &compile_config()` to ` const CompileConfig &compile_config()`. Please review again.